app/scripts/controllers/visualization.js

```diff
@@ -1,66 +1,15 @@
 'use strict';
 
-/**
- * @ngdoc function
- * @name trunkApp.controller:VisualizationCtrl
- * @description
- * # VisualizationCtrl
- * Controller of the trunkApp
- */
-app.controller('VisualizationCtrl', ['$scope', 'ViewerFactory', 'DatasetService', function ($scope, ViewerFactory, DatasetService) {
-
-  var fetchGenotype = function() {
-    var foram = DatasetService.getProducts()[0];
-
-    if (foram) {
-      return foram.genotype;
-    } else {
-      return null;
-    }
-  };
-
-  // initialize our ViewerFactory responsible for 3D visualization
-  var viewer = new ViewerFactory({
-    containerId: '#WebGL-output',
-    genotype:    fetchGenotype()
-  });
+app.controller('VisualizationCtrl', ['$scope', 'SimulationFactory', 'GenotypeService', function ($scope, simulationFactory, genotypeService) {
+  var simulation = simulationFactory($('#visualization'));
+  simulation.animate();
 
   $scope.data = {
-    'scale': 1,
-    'rotateX': 0,
-    'rotateY': 0,
-    'rotateZ': 0,
-    'positionX': 0,
-    'positionY': 0,
-    'positionZ': 0
-  };
-
-  // scale the model.
-  $scope.scale = function () {
-    viewer.scale(this.data.scale);
-  };
-
-  // rotate around an axis
-  $scope.rotate = function () {
-    viewer.rotate(
-      parseFloat(this.data.rotateX),
-      parseFloat(this.data.rotateY),
-      parseFloat(this.data.rotateZ))
-  };
-
-  // translate around the scene
-  $scope.translate = function () {
-    viewer.translate(
-      parseFloat(this.data.positionX),
-      parseFloat(this.data.positionY),
-      parseFloat(this.data.positionZ))
-  };
-
-  $scope.evolve = function() {
-    viewer.evolve();
+    genotype: genotypeService.fetchGenotype(),
+    options:  { numChambers: 7 }
```

Comment:
Trochę zastanawia mnie ta siódemka - rozumiem że to wybrana przez Ciebie domyślna wartość. Raczej bym wyciągnął to do jakiegos pliku konfiguracyjnego i pobrał z serwisu do configów. Ale możemy to przedyskutować :)